src/lib/server/database/index.ts

```diff
@@ -47,12 +51,15 @@ export async function insertConfession(
     description: string,
     approvedAt: Date | null,
     parentMessageId: Snowflake | null,
+    attachmentUrl: Attachment['url'] | null,
+    attachmentFilename: Attachment['filename'] | null,
+    attachmentType: Attachment['content_type'] | null,
```

Comment:
This can be improved by accepting an `Attachment | null` object as a whole. From an interface standpoint, I find it highly unlikely that we would even care about partially populated attachment fields.

Reply:
Okay, per your recommendation on the database model I'll set up a new table for storing whole `attachment` objects with `confession` storing a foreign key reference to it.

Reply:
This has been resolved in the latest commits.